aspnet/mobile/native-mobile-backend.rst

```diff
@@ -1,11 +1,170 @@
+
+The application should respond to all requests made to port 5000. Update `Program.cs` to include ``.UseUrls("http://*:5000")`` to achieve this:
+
+.. literalinclude:: native-mobile-backend/sample/ToDoApi/src/ToDoApi/Program.cs
+  :language: c#
+  :lines: 10-16
+  :dedent: 12
+  :emphasize-lines: 3
+
+.. note:: Make sure you run the application using Kestrel, not IIS Express. This can be done from Visual Studio, or by running ``dotnet run`` from the command line.
+
+Add a model class to represent To-Do items. Mark required fields using the ``[Required]`` attribute:
+
+.. literalinclude:: native-mobile-backend/sample/ToDoApi/src/ToDoApi/Models/TodoItem.cs
+  :language: c#
+
+The API methods require some way to work with data. Use the same ITodoRepository interface the original Xamarin sample uses:
+
+.. literalinclude:: native-mobile-backend/sample/ToDoApi/src/ToDoApi/Interfaces/ITodoRepository.cs
+  :language: c#
+
+For this sample, the implementation just uses a private collection of items. Configure the implementation in `Startup.cs`:
+
+.. literalinclude:: native-mobile-backend/sample/ToDoApi/src/ToDoApi/Startup.cs
+  :language: c#
+  :lines: 30-36
+  :dedent: 8
+  :emphasize-lines: 6
+  
+At this point, you're ready to create the `TodoItemsController`.
+
+Creating the Controller
+-----------------------
+
+Add a new controller to the project, `TodoItemsController`. It should inherit from Microsoft.AspNetCore.Mvc.Controller. Add a ``Route`` attribute to indicate that the controller will handle requests made to paths starting with ``api/todoitems``, which may optionally include an ``id`` argument as part of the URL path. The ``[controller]`` token in the route is replaced by the name of the controller (omitting the ``Controller`` suffix), and is especially helpful for global routes. Learn more about :doc:`routing </fundamentals/routing>`.
+
+The controller requires an ``ITodoRepository`` to function; request an instance of this type through the controller's constructor. At runtime, this instance will be provided using the framework's support for :doc:`dependency injection </fundamentals/dependency-injection>`.
+
+.. literalinclude:: native-mobile-backend/sample/ToDoApi/src/ToDoApi/Controllers/TodoItemsController.cs
+  :language: c#
+  :lines: 1-17
+
+This API supports four different HTTP verbs to perform CRUD (Create, Read, Update, Delete) operations on the data source. The simplest of these is the Read operation, which corresponds to an HTTP GET request.
+
+Reading Items
+^^^^^^^^^^^^^
+
+Requesting a list of items is done with a GET request to the List() method. The method name doesn't matter as long as it's the only method matching the request for the given HTTP verb (in this case, GET). If there were multiple GET methods, it would be necessary to add ``[Route]`` attributes to each action (or add ``[action]`` to the route, which would use the action's name as part of the route).
```

Comment:
You should also explicitly state that returning `Ok(_todoRepository.All)` will return the todo items serialized as JSON with a 200 OK response code.